core/docs/error-handling.md

```diff
@@ -0,0 +1,23 @@
+## Error handling
+
+PureConfig provides some semantically rich case classes to describe failures
```

Comment:
Suggestion: I'd reword this sentence as:

> The `pureconfig.error` package provides semantically rich case classes to describe failures reading from a `ConfigValue`.

Feel free to prefer your wording.